### ISSUE TITLE
Integrate some part of fractal-demos

### ISSUE THREAD

**Issue by tcompa** [fractal-analytics-platform/fractal-containers#18]
Related to https://github.com/fractal-analytics-platform/fractal-demos/issues/37.

Proposal: let's set up one more container with fractal-client, and let's have it as ready as possible to run example 01 from fractal-demos. Ideally all should be automated, meaning that `make run` should trigger a bunch of operations that are over only once example 01 is over.

Note that there will be a few expected gotcha on the way: e.g. paths need to be updated to something in `/home/fractal-share`, and we should wait for task collection to be over (through a script that checks every N seconds), ...

Longer-term perspective: this could represent one way to automatically re-run a bunch of fractal-demos examples. TBD what kind of checks could be included in the output (either just successful-execution assertions, or something more related to the output data - ref https://github.com/fractal-analytics-platform/fractal-demos/issues/37).

**Comment by tcompa** [comment]
A broader review, planned with @mfranzon.

## Versions

Expose env variables for versions (all defaulting to latest pypi version, if not specified) of
* fractal-server
* factal-tasks-core
* fractal-client
* fractal-web (latest release on github)
* fractal-demos (maybe either latest on github or specific commit - to be used with `git checkout`)

Each variable gets used in the right container.

## Server container and DB container

* Use versions
* Make volumes non persistent.

## Web client

* Use versions

## demos container

* Use versions (both in Dockerfile and `run_all_demos.sh`)
* Move as many statements as possible from `run_all_demos.sh` to Dockerfile.
* Pip install a minimal set of requirements for `check_result.py`
* Split `run_all_demos.sh` in setup (up to task collection is over), example01, .. -> and/or collect all in a single script..
* https://github.com/fractal-analytics-platform/fractal/issues/552
* Catch exit code of run_example and log its state somewhere
* After `run_example.sh`, add a new layer that waits for when job is over
* Catch job status and log it somewhere
* Add an example Python script for checking output, based on https://github.com/fractal-analytics-platform/fractal-demos/issues/37
* Run `check_results.py` and log its output somewhere

Expected: a log file like
```
example 01
API calls -> success (or failed, here is the log..)
Job status -> done (or failed, here is the log..)
check-results -> success (or failed, here is the log..)
```

## github action

* Add all of this as a github action (without setting versions, in the first version), and use `workflow_dispatch`

**Comment by tcompa** [comment]
Here is a proposal about versions:

For `fractal-client` (and the same holds for `fractal-server`) we may have one of the following env variable: `FRACTAL_CLIENT_RELEASE_VERSION` or `FRACTAL_CLIENT_GIT_VERSION`. The logic could be like:
1. If they are both unset, then we use `pip install fractal-client`.
2. If they are both set, we fail.
3. If `FRACTAL_CLIENT_RELEASE_VERSION` is set, then we use it as part of a `pip install` command.
4. If `FRACTAL_CLIENT_GIT_VERSION` is set (this could be a commit hash, or a tag, or a branch), then we go through git-clone and then git-checkout.

For `fractal-tasks-core`, we have fewer cases:
1. If `FRACTAL_TASKS_CORE_RELEASE_VERSION` is set, we pass it to the fractal-task-collect command.
2. If it is not set, we do not include it in the fractal-task-collect command.

For fractal-web, we could stick with `FRACTAL_WEB_GIT_VERSION` (a commit hash, a branch, or a tag):
1. If it's unset, we git-clone and stick with `main`
2. If it's set, we git-clone and then git-checkout.

**Comment by tcompa** [comment]
> For this, we would need some logic to shutdown everything after the demos service is over.

See `--abort-on-container-exit` flag

**Comment by jluethi** [comment]
> For fractal-tasks-core, we have fewer cases:
> 
> If FRACTAL_TASKS_CORE_RELEASE_VERSION is set, we pass it to the fractal-task-collect command.
> If it is not set, we do not include it in the fractal-task-collect command.

Wouldn't we want to be able to use this to e.g. test a new task version or a task branch? Or is the focus on testing changes to server & web server with the released tasks?

**Comment by tcompa** [comment]
Closed with #21
